src/reader.rs

```diff
@@ -4971,7 +4973,7 @@ fn reader_run_command(parser: &Parser, cmd: &wstr) -> EvalRes {
         );
     }
 
-    term_steal();
+    term_steal(eval_res.status.is_success());
```

Comment:
actually your suggestion of copying modes only on success makes a lot of sense.
Do you maybe have a concrete example of bad modes copied, was it something from ssh too?

Reply:
>  was it something from ssh too?
Yes.

Reply:
ok but was it something that is fixed by `term_fix_external_modes()`? Since that's been re-added in latest master.
It's not very clearly defined what we should do here so I'd prefer a concrete motivation (like `ECHO` not being set or similar)

Reply:
The `term_fix_external_modes()` function is incomplete. I attempted to manually restore additional options like `echoe` and `echok`. However, some options, such as `ignpar` and `ignbrk`, don't even exist in the rust libc package. Additionally, I'm unsure if more options need to be restored. Using `stty sane` results in a different set of options compared to the initialization of fish shell, which is why I opted for a different approach instead.